gui/widgets/tabs.py

```diff
@@ -97,38 +122,50 @@ def __init__(self, num_video_streams):
         self.current_filter = "None"  # Filter applied with dropdown menu
 
         # Creating combo_box and adding the functions
-        self.combo_box = QComboBox()
+        combo_box = QComboBox()
 
         for func_name in dropdown.func_dictionary.keys():
-            self.combo_box.addItem(func_name)
+            combo_box.addItem(func_name)
 
-        self.combo_box.currentTextChanged.connect(self.update_current_filter)
-        self.update_current_filter(self.combo_box.currentText())
+        combo_box.currentTextChanged.connect(self.update_current_filter)
+        self.update_current_filter(combo_box.currentText())
 
-        self.root_layout.addWidget(self.combo_box)
+        self.widgets.filter_dropdown = combo_box
 
         # Add video control buttons
-        self.video_controls = VideoControlsWidget()
-        self.root_layout.addWidget(self.video_controls)
+        video_controls = VideoControlsWidget()
+        self.widgets.video_controls = video_controls
 
         # Add select files button
-        self.select_files_button = QPushButton(self)
-        self.select_files_button.setText("Select Files")
-        self.select_files_button.clicked.connect(self.select_files)
-        self.root_layout.addWidget(self.select_files_button)
+        select_files_button = QPushButton(self)
+        select_files_button.setText("Select Files")
+        select_files_button.clicked.connect(self.select_files)
+        self.widgets.select_files_button = select_files_button
```

Comment:
```suggestion
        self.widgets.select_files_button = select_files_button
        self.organize()
```

Reply:
Yes, it would, we'd just have to have to make sure that the constructor of every tab that will be directly instantiated ends with a call to `self.organize`. We wouldn't be able to subclass those classes without `organize` being called twice, which might screw up the image debug tab PR, I haven't checked how it's implemented. With that said I don't think we're going to be creating many more tabs so that probably wouldn't be a big deal. Up to you on which way you think is cleaner.

Reply:
Another approach could be having two methods each tab class must implement: `init_widgets` and `organize`. So we'd have:
```Python
class RootTab(QWidget):
    def __init__(self):
        super().__init__()
        self.init_widgets()
        self.organize()

class ChildTab(RootTab):
    def __init__(self):
        super().__init__()

    def init_widgets(self):
        super().init_widgets()
        # Init other widgets

    def organize(self):
        super().organize()
        # Organize other widgets
```

Reply:
Hmm, yeah, good idea. I'll try adding that.

Reply:
Just implemented this in 7ea4460